### PR TITLE
[TS-711]<fix>: windows shell limit 254 char error.

### PR DESCRIPTION
**Pull request by afwerar**

### CODE REVIEW

File: src/kit/shell/src/shellWindows.c [hunk 1]
```diff
@@ -274,32 +276,34 @@ int32_t shellReadCommand(TAOS *con, char command[]) {
   // Read input.
   void *console = GetStdHandle(STD_INPUT_HANDLE);
   unsigned long read;
-  wchar_t c;
+  wchar_t c[SHELL_INPUT_MAX_COMMAND_SIZE];
```
Comment (sangshuduo) [inline src/kit/shell/src/shellWindows.c]: how about malloc()/free() instead of using an array on stack?

Comment (afwerar) [reply src/kit/shell/src/shellWindows.c]: ok

**Review by stephenkgu — APPROVED** [review]
good fix

File: src/kit/shell/src/shellWindows.c [hunk 2]
```diff
@@ -274,32 +276,35 @@ int32_t shellReadCommand(TAOS *con, char command[]) {
...
+            free(cmd.command);
+            cmd.command = NULL;
+            free(c);
+            return 0;
+          } else {
+            shellPrintContinuePrompt();
+            updateBuffer(&cmd);
+          }
+          break;
+        case '\r':
```
Comment (sangshuduo) [inline src/kit/shell/src/shellWindows.c:301]: array 'c' be freed in case '\n' only, but seems the memory is not freed if the loop break from 'r'?

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**